streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1200,7 +1179,7 @@ private long getCacheSizePerThread(final int numStreamThreads) {
         if (numStreamThreads == 0) {
             return totalCacheSize;
         }
-        return totalCacheSize / (numStreamThreads + ((globalTaskTopology != null) ? 1 : 0));
+        return totalCacheSize / (numStreamThreads + (topologyMetadata.hasGlobalTopology() ? 1 : 0));
```

Comment:
If we have more than one named topologies that have global stores, should the global topologies share one cache sub-space?

Reply:
At the moment we just don't allow global stores with named topologies. There is a list of not-yet-supported features that are currently incompatible with them in the javadocs of KafkaStreamsNamedTopologyWrapper

Reply:
We're just trying to get the basic NamedTopology feature up and running for the time being, so we'll loop back around on any "missing" features eventually.